sdk/include/opentelemetry/sdk/common/http_client.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <chrono>
+#include <string>
+#include <vector>
+#include "opentelemetry/nostd/function_ref.h"
+#include "opentelemetry/version.h"
+
+// Usage Example
+//
+// struct ReponseHandler: public HttpResponseHandler {
+//      void OnHttpResponse(HttpResponse& res) noexcept override
+//      {
+//           if (res.IsSuccess())
+//              .. process response
+//      }
+//  };
+//
+//   HttpSessionManager sessionManager; // implementer can provide singleton implementation for it
+//   auto session = sessionManager.createHttpSession("localhost", 8000);
```

Comment:
nit: is `8000` a random choice here?

Reply:
It seems `createHttpSession` could just be static method. Is there any benefit to create a even a singleton instance?

Reply:
Yes you are right. In fact, all the methods of `HttpSessionManager` class can be static. I just didn't want to impose that as part of API. As it provides flexibility to have different `HttpSessionManager` for each exporters, or to create single global singleton instance as per the need.

Reply:
> nit: is `8000` a random choice here?

Yes, kind of - The http server I was using for testing was listening to this port, so the example.